core/state.go

```diff
@@ -397,11 +398,8 @@ func (s *State) updateContractStorages(stateTrie *trie.Trie, diffs map[felt.Felt
 
 	// sort the contracts in decending diff size order
 	// so we start with the heaviest update first
-	keys := make([]felt.Felt, 0, len(diffs))
-	for key := range diffs {
-		keys = append(keys, key)
-	}
-	slices.SortStableFunc(keys, func(a, b felt.Felt) int { return len(diffs[a]) - len(diffs[b]) })
+	mapKeys := maps.Keys(diffs)
+	keys := slices.SortedStableFunc(mapKeys, func(a, b felt.Felt) int { return len(diffs[a]) - len(diffs[b]) })
```

Comment:
Minor nit pick:

`mapKeys` are not used anywhere, might be better to just do

```
keys := slices.SortedStableFunc(maps.Keys(diffs), func(a, b felt.Felt) int { return len(diffs[a]) - len(diffs[b]) })
```

Reply:
done